src/test/java/mmosii/bookstore/service/ShoppingCartServiceTest.java

```diff
@@ -0,0 +1,257 @@
+
+        ShoppingCart shoppingCart = new ShoppingCart();
+        shoppingCart.setUser(user);
+        shoppingCart.setId(3L);
+
+        ShoppingCartDto expected = new ShoppingCartDto(shoppingCart.getId(),
+                shoppingCart.getUser().getId(), null);
+
+        when(securityContext.getAuthentication()).thenReturn(authentication);
+        when(authentication.getName()).thenReturn("test@gmail.com");
+        when(userRepository.findByEmail("test@gmail.com")).thenReturn(Optional.of(user));
+        when(shoppingCartRepository.findByUserId(user.getId()))
+                .thenReturn(Optional.of(shoppingCart));
+        when(shoppingCartMapper.toDto(shoppingCart)).thenReturn(expected);
+
+        ShoppingCartDto actual = shoppingCartService.getShoppingCart();
+
+        assertThat(actual).isEqualTo(expected);
+        verify(securityContext, times(1)).getAuthentication();
+        verify(authentication, times(1)).getName();
+        verify(userRepository, times(1)).findByEmail("test@gmail.com");
+        verify(shoppingCartRepository, times(1)).findByUserId(user.getId());
+        verify(shoppingCartMapper, times(1)).toDto(shoppingCart);
+        verifyNoMoreInteractions(securityContext, authentication, userRepository,
+                shoppingCartRepository, shoppingCartMapper);
+    }
+
+    @Test
+    @DisplayName("Get new shopping cart")
+    void getShoppingCart_newCart_returnsCartDto() {
+        SecurityContextHolder.setContext(securityContext);
+        User user = new User();
+        user.setId(5L);
+        user.setEmail("test@gmail.com");
+
+        ShoppingCart shoppingCart = new ShoppingCart();
+        shoppingCart.setUser(user);
+
+        ShoppingCartDto expected = new ShoppingCartDto(shoppingCart.getId(),
+                shoppingCart.getUser().getId(), null);
+
+        when(securityContext.getAuthentication()).thenReturn(authentication);
+        when(authentication.getName()).thenReturn("test@gmail.com");
+        when(userRepository.findByEmail("test@gmail.com")).thenReturn(Optional.of(user));
+        when(shoppingCartRepository.findByUserId(user.getId())).thenReturn(Optional.empty());
+        when(shoppingCartRepository.save(any())).thenReturn(shoppingCart);
+        when(shoppingCartMapper.toDto(shoppingCart)).thenReturn(expected);
+
+        ShoppingCartDto actual = shoppingCartService.getShoppingCart();
+
+        assertThat(actual.userId()).isEqualTo(expected.userId());
+        verify(securityContext, times(1)).getAuthentication();
+        verify(authentication, times(1)).getName();
+        verify(userRepository, times(1)).findByEmail("test@gmail.com");
+        verify(shoppingCartRepository, times(1)).findByUserId(user.getId());
+        verify(shoppingCartRepository, times(1)).save(any());
+        verify(shoppingCartMapper, times(1)).toDto(shoppingCart);
+        verifyNoMoreInteractions(securityContext, authentication,
+                userRepository, shoppingCartRepository, shoppingCartMapper);
```

Comment:
From mockito documentation:

> A word of warning: Some users who did a lot of classic, expect-run-verify mocking tend to use verifyNoMoreInteractions() very often, even in every test method. verifyNoMoreInteractions() is not recommended to use in every test method. verifyNoMoreInteractions() is a handy assertion from the interaction testing toolkit. **Use it only when it's relevant. Abusing it leads to over-specified, less maintainable tests.**

And also about verify:

> **Although it is possible to verify a stubbed invocation, usually it's just redundant.** Let's say you've stubbed foo.bar(). If your code cares what foo.bar() returns then something else breaks(often before even verify() gets executed). If your code doesn't care what foo.bar() returns then it should not be stubbed.

I think this methods more useful for testing void methods. So you can skip this comment or think to delete them (from Nazar's words) :)